e2e/test/iothub/properties/PropertiesE2ETests.cs

```diff
@@ -0,0 +1,448 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Azure.Devices.E2ETests.Helpers;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Devices.E2ETests.Properties
+{
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class PropertiesE2ETests : E2EMsTestBase
+    {
+        private readonly string _devicePrefix = $"E2E_{nameof(PropertiesE2ETests)}_";
+
+        private static readonly RegistryManager s_registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+        private static readonly TimeSpan s_maxWaitTimeForCallback = TimeSpan.FromSeconds(30);
+
+        private static readonly Dictionary<string, object> s_mapOfPropertyValues = new Dictionary<string, object>
+        {
+            { "key1", 123 },
+            { "key2", "someString" },
+            { "key3", true }
+        };
+
+        [LoggedTestMethod]
+        public async Task Properties_DeviceSetsPropertyAndGetsItBack_Mqtt()
```

Comment:
I know we follow this style of tests in our code. But for the future something to consider for better readability of e2e tests is to add sections for `arrange` `act` `assert` and use [Theory] with multiple protocol inputs. It is very hard to understand what is being setup, what is being called and checked for like this. There might be some amount of repeated code with the arrange, act, assert format but each test will be very readable.

Reply:
I agree. I've found unit tests to be very easy to use the arrange-act-assert format, but since end-to-end tests usually have these actions spread out within the test, I've found those comments unnecessarily cluttering up the test. We can try it out in a couple of tests and see how we feel.

Reply:
I haven't used `[Theory]` before, is that like parameterized tests?

Reply:
yupp those are parameterized tests. Theory is in xunit but it is called datarow in ms test Ex of something I added recently - https://github.com/Azure/azure-iot-sdk-csharp/blob/54e8280f775f7c6db00a29b98169588532180fc2/e2e/test/iothub/service/RegistryManagerExportDevicesTests.cs#L45

I feel writing tests using 'arrange act assert' makes it very readable and is a commonly followed format.

Reply:
Ah, I see. Parameterized MS tests don't support parallelization yet, so we have refrained from using those in our E2E tests suite since it'll end up increasing the test run time.